src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -125,10 +125,48 @@ public SelectExpandClause Bind(ExpandToken expandToken, SelectToken selectToken)
             {
                 // if there are any select items at this level then allSelected is false, otherwise it's true.
                 isAllSelected = false;
+                HashSet<PathSegmentToken> selectedPaths = new HashSet<PathSegmentToken>(new PathSegmentTokenEqualityComparer());
 
                 foreach (SelectTermToken selectTermToken in selectToken.SelectTerms)
                 {
-                    AddToSelectedItems(GenerateSelectItem(selectTermToken), selectExpandItems);
+                    SelectItem selectItem = GenerateSelectItem(selectTermToken);
+                    if (selectedPaths.Contains(selectTermToken.PathToProperty))
+                    {
+                        // It's not allowed have multple select clause with same path.
+                        // For example: $select=abc($top=2),abc($skip=2) is not allowed by design.
+                        // Customer should combine them together, for example: $select=abc($top=2;$skip=2).
+                        // The logic is different with ExpandTreeNormalizer. We should change the logic in ExpandTreeNormalizer
+                        // in next breaking change version.
+                        // For backward compatibility with previous versions of OData Library, we only validate
+                        // if one of the select item has options.
+                        PathSelectItem selectPathItem = selectItem as PathSelectItem;
+
+                        // if the new one has options, raise an error
+                        if (selectPathItem != null && selectPathItem.HasOptions)
+                        {
+                            throw new ODataException(ODataErrorStrings.SelectTreeNormalizer_MultipleSelecTermWithSamePathFound(ToPathString(selectTermToken.PathToProperty)));
+                        }
+                        else
+                        {
+                            // find the previous item
+                            PathSelectItem previousItem = selectExpandItems.Cast<PathSelectItem>().FirstOrDefault(selectExpandItem =>

```

Comment:
maybe it's better to use `OfType<T>() `, not `cast<T>()`. `OfType<PathSelectItem().FirstOrDefault(...)`

Reply:
I was ignoring instances that weren't PathSelectItem in my filter. Not sure which is more efficient, but using cast is certainly simpler and may avoid an extra cast.

---
In reply to: [340909397](https://github.com/OData/odata.net/pull/1583#discussion_r340909397) [](ancestors = 340909397)